tests/h/auth/util_test.py

```diff
@@ -200,6 +202,114 @@ def test_does_not_raise_when_authority_matches(self, pyramid_request, auth_clien
         util.validate_auth_client_authority(auth_client, authority)
 
 
+class TestPrincipalsForAuthClient(object):
```

Comment:
These tests were quite challenging to write because if you look at [the source for this function](https://github.com/hypothesis/h/pull/5192/files#diff-f54a90f529b1f3f68d72d8c0743466acR146) you'll see that there are a goodly handful of conditions that will cause it to return `None`, meaning the challenge is to mock in such a way that the individual `None` conditions are testable. Not sure if I hit the mark here, but I tried.